skbio/tree/_tree.py

```diff
@@ -3098,3 +3098,33 @@ def shuffle(self, k=None, names=None, shuffle_f=np.random.shuffle, n=1):
 
             yield self
             counter += 1
+
+    def _cache_leafcount(self):
```

Comment:
I don't think that'll work, since there can be un-named nodes that have branch length.